frontend/src/components/utils/Authentication.tsx

```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+import { useEffect } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+
+const authenticatedRouts = ['/home'];
+const unauthenticatedRouts = ['/login', '/registration'];
+
+const Authentication = ({children}: any) => {
+    const history = useHistory();
+    const location = useLocation();
+
+    useEffect(() => {
+        let isAuthenticated = false;
+
+        const getUserRoute = () => {
+            let currentRoute;
+            let defaultRoute = '/login';
+
+            if(isAuthenticated) {
+                defaultRoute = '/home';
+                currentRoute = authenticatedRouts.find(route => route === location.pathname);
+            } else {
+                defaultRoute = '/login';
+                currentRoute = unauthenticatedRouts.find(route => route === location.pathname);
+            }
+
+            if(currentRoute) {
+                history.push(currentRoute);
+            } else {
+                history.push(defaultRoute);
+            }
+        }
+
+        try {
+            if(!localStorage.getItem('token')) {
+                isAuthenticated = false;
+                getUserRoute();
```

Comment:
You can redirect user to `/login` here, no need to call `getUserRoute`.

Reply:
Then the user won't be able to access the **/registration** page.